libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -679,6 +679,10 @@ bool AC_PrecLand::construct_pos_meas_using_rangefinder(float rangefinder_alt_m,
     return false;
 }
 
+float AC_PrecLand::get_target_distance_m() {
+    return _backend->distance_to_target();
```

Comment:
`get_target_position_measurement_cm` returns a 3-D Vector. The "Z" component of this vector is either filled by the rangefinder or the distance specified by Mavlink. Can't we just use that?

Reply:
Hi @rishabsingh3003 Thank you very much for suggestions. I did not notice this function (`get_target_position_measurement_cm`).  I will change to use it and do flight test.

Reply:
Hi @rishabsingh3003 I think `construct_pos_meas_using_rangefinder` already take care of rangefinder reading. Maybe we can simply use 3d vector from `get_target_position_measurement_cm` only?
https://github.com/ArduPilot/ardupilot/blob/8bd26dffaccf4dc35c0c0035801eb2bca8f1b811/libraries/AC_PrecLand/AC_PrecLand.cpp#L657-L659

Reply:
@chobitsfan yes I think so (that's what my original suggestion was). You will have to test but it'll greatly simplify things. I like the idea of a distance coming from the PrecLand library as it gives us more flexibility in the future to handle complex things from within PrecLand.

Reply:
>  You will have to test but it'll greatly simplify things.

Yes, I will test it once weather condition is ok (maybe 2 or 3 days later, due to typhoon)